### PR TITLE
feat: Setup Common Config

### PR DESCRIPTION
**Pull request by doomspork**
WIP

### CODE REVIEW

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Sync Repositories
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
+          script: |
+            const { data } = await github.rest.search.code({
+              q: 'Common Config org:beam-community path:.github/workflows/',
```
Comment (doomspork) [inline .github/workflows/publish.yaml:27]: @btkostner thoughts on how to makes this extensible to more than 1 org? 🤔

File: .github/workflows/pr.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        uses: actions/checkout@v3
+
+      - name: Check
+        uses: stordco/actions-pr-title@v1.0.0
+        with:
+          regex: '^(feat!|fix!|fix|feat|chore|(fix|feat|chore)\(\w.*\)):\s(\[\w{1,8}-\d{1,8}\]|.*).*'
+          hint: |
+            You can pass the following formats:
+
+            fix: [OR-123] some title of the PR
```
Comment (btkostner) [inline .github/workflows/pr.yaml:28]: These should be changed to resemble GitHub issues

File: .github/workflows/publish.yaml (hunk 1 above)
Comment (btkostner) [reply .github/workflows/publish.yaml:27]: No clue but there is probably some GH query for it. One thing to note is that the auth needs to have push permissions. So this dispatch works for beam org, all other orgs will just have to run via cron

Comment (btkostner) [inline .github/workflows/publish.yaml:27]: We should be able to just query for the common-config.yaml file

File: README.md [hunk 3]
```diff
@@ -1 +1,19 @@
...
+
+<!-- {x-release-please-start-version} -->
+
+Currently at version `1.0.0`
+
+<!-- {x-release-please-end} -->
+
+## Setting up repositories
+
+For information about setting up other repositories to sync with common-config, read the [GitHub Action section of the documentation](https://elixir.stord.engineering/action.html).
```
Comment (btkostner) [inline README.md:13]: Doc link needs updating

File: templates/.github/workflows/ci.yaml [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    types:
+      - opened
+      - reopened
+      - synchronize
+  push:
+    branches:
+      - main
+  workflow_call:
+    secrets:
+      CI_SERVICE_KEY:
```
Comment (btkostner) [inline templates/.github/workflows/ci.yaml:17]: This key is not required

File: templates/.github/workflows/ci.yaml [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_INITDB_ARGS: "--nosync"
+
+{{/if}}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Elixir
+        uses: stordco/actions-elixir/setup@v1
```
Comment (btkostner) [inline templates/.github/workflows/ci.yaml:121]: I think this action has some stord specific configuration that will need updating

File: templates/.github/workflows/pr.yaml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    steps:
+      - name: Check
+        uses: stordco/actions-pr-title@v1.0.0
+        with:
+          regex: '^(feat!|fix!|fix|feat|chore|(fix|feat|chore)\(\w.*\)):\s(\[\w{1,8}-\d{1,8}\]|.*).*'
+          hint: |
+            Your PR title does not match the Stord common convention. Please rename your PR to match one of the following formats:
+
+            chore: a small insignificant change
+            fix: [JIRA-1234] fix an existing feature
```
Comment (btkostner) [inline templates/.github/workflows/pr.yaml:29]: We don't use Jira

File: templates/.github/workflows/production.yaml [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Elixir
+        uses: stordco/actions-elixir/setup@v1
+        with:
+          github-token: $\{{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
+          hex-token: $\{{ secrets.HEX_API_KEY }}
+          oban-fingerprint: $\{{ secrets.OBAN_KEY_FINGERPRINT }}
+          oban-token: $\{{ secrets.OBAN_LICENSE_KEY }}
```
Comment (btkostner) [inline templates/.github/workflows/production.yaml:30]: We don't use oban pro

**Review by btkostner — CHANGES_REQUESTED** [review]

File: .github/workflows/publish.yaml (hunk 1 above)
Comment (doomspork) [reply .github/workflows/publish.yaml:27]:
```suggestion
              q: 'org:beam-community path:.github/workflows/common-config.yaml',
```

**Review by btkostner — APPROVED** [review]

**Pull request merged**